### PR TITLE
PCHR-2025: Allow admin edit all fields in leave request popup

### PR DESCRIPTION
**Pull request by igorpavlov**
## Overview

Admin is now able to edit all fields in the popup introduced in https://github.com/civicrm/civihr/pull/1967.

## After

new

## Technical Details

In https://github.com/civicrm/civihr/pull/1967 the `this.canManage` was introduced to equalise admin's rights with manager's rights in the popup. In this PR admin has *more* rights, than manager, so pieces of code with `this.canManage` _*plus* extra conditions_ were amended to ensure that admin is granted with access disregarding these extra conditions. Where no extra conditions were presented, `this.canManage` was left.

## Comments

As a part of this PR date pickers in the popup were improved:
- Transparent background colour _(only to Admin)_
- Default not opaque text _(only to Admin)_
- Clicking on text inputs calls out the date picking calendar (both SSP and Admin)

Also, the staff search bar (selector) is now hidden in edit mode because it is not needed as the contact is already specified in the header and cannot be changed.

---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss [hunk 1]
```diff
@@ -129,4 +129,11 @@ $row-margin-bottom: 15px;
   .alert {
     margin-bottom: 0 !important;
   }
+
+  input[uib-datepicker-popup] {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss]: explain why in a comment

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js [hunk 2]
```diff
@@ -1012,6 +1012,10 @@ define([
        * @return {Promise}
        */
       function loadManagees () {
+        if (this.isMode('edit')) {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js]:
```js
return self.canManage && !this.isMode('edit') && loadManagees.call(self);

Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js]: Done.

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss (hunk 1 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss]: Done.

**Pull request merged**